app/src/main/java/com/nextcloud/talk/utils/DateUtils.kt

```diff
@@ -41,6 +54,48 @@ object DateUtils {
     }
 
     fun getLocalDateStringFromTimestampForLobby(timestamp: Long): String {
-        return getLocalDateTimeStringFromTimestamp(timestamp * 1000)
+        return getLocalDateTimeStringFromTimestamp(timestamp * TIMESTAMP_CORRECTION_MULTIPLIER)
+    }
+
+    fun relativeStartTimeForLobby(timestamp: Long, resources: Resources): String {
+
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            val fmt = RelativeDateTimeFormatter.getInstance()
+            val timeLeftMillis = timestamp * TIMESTAMP_CORRECTION_MULTIPLIER - System.currentTimeMillis()
+            val minutes = timeLeftMillis / SECOND_DIVIDER / MINUTES_DIVIDER
+            val hours = minutes / HOURS_DIVIDER
+            val days = hours / DAYS_DIVIDER
+
+            val relative = when {
+                days > 0 -> {
+                    fmt.format(
+                        days.toDouble(),
+                        Direction.NEXT,
+                        RelativeUnit.DAYS
+                    )
+                }
+                hours > 0 -> {
+                    fmt.format(
+                        hours.toDouble(),
+                        Direction.NEXT,
+                        RelativeUnit.HOURS
+                    )
+                }
+                minutes > 0 -> {
+                    fmt.format(
+                        minutes.toDouble(),
+                        Direction.NEXT,
+                        RelativeUnit.MINUTES
+                    )
+                }
+                else -> {
+                    resources.getString(R.string.nc_lobby_start_soon)
```

Comment:
the soon case should not be done the same way the relative date works. As Marcel pointed out it breaks with translations

Reply:
> the soon case should not be done the same way the relative date works. As Marcel pointed out it breaks with translations

You're right, of course. I'll change it.

Reply:
I left the soon case in the else part of the relative dates handling but append the complete text in the lobby screen in an other way.